### PR TITLE
Add profiler

### PR DESCRIPTION
**Pull request by Benny-Nottonson**
This pull request adds a profiler to the codebase. The profiler allows for performance analysis and generates a flame graph visualization.

### CODE REVIEW

**Comment by StijnWoestenborghs** [comment]
Are you on WSL for this? As I'm on Linux and if we are going to add a profiler to the main branch we should make it work for all platforms probably. The changes I had to make were:
```
... "$temp_dir/run.exe" --> "$temp_dir/run"
~/perf -> sudo perf 
explorer.exe --> google-chrome $flamegraph_output
```

I'll have a look later, for the script to recoginze the platform. 
@andresnowak  Are you on Mac perhaps?

**Comment by Benny-Nottonson** [comment]
Good point about being platform independent, im on WSL2 Ubuntu, I also have a Mac I can test on if needed, but I agree, we should unify that for all platforms. The ~/perf is a result of a workaround needed to install perf on linux, and could probably be put into a seperate step, ill look into detecting platform.

**Comment by Benny-Nottonson** [comment]
@StijnWoestenborghs Check now :)

File: profile.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    echo "Building $mojo_file..."
+    mojo build -O0 -I . "$mojo_file"
+
+    echo "Stripping debug symbols..."
+    mv "$mojo_name" "$temp_dir/run.exe"
+    llvm-strip --strip-debug "$temp_dir/run.exe"
+
+    echo "Running perf record..."
+    perf record -F 99 -a -g -o "$perf_output" -- "$temp_dir/run.exe"
```
Comment (StijnWoestenborghs) [inline profile.sh:55]: For some reason i still need a sudo here to get it to work. 
And does it work for you without the .exe (=windows specific)

File: profile.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    echo "Generating flamegraph..."
+    perf script -i "$perf_output" | ~/FlameGraph/stackcollapse-perf.pl | ~/FlameGraph/flamegraph.pl > "$flamegraph_output"
```
Comment (StijnWoestenborghs) [inline profile.sh:58]: same here, needs sudo

**Pull request merged**